runtime/src/bank.rs

```diff
@@ -7803,6 +7803,17 @@ impl Bank {
             &mut error_counters,
         )
     }
+
+    pub fn is_in_slot_hashes_history(&self, slot: &Slot) -> bool {
+        if slot < &self.slot {
+            if let Ok(sysvar_cache) = self.sysvar_cache.read() {
```

Comment:
Double checked and there shouldn't be any contention on this, as only replay should be using it when executing transactionw